generate_isoscape.py

```diff
@@ -0,0 +1,201 @@
+
+  # Initialize a blank plane representing means and variance.
+  predicted_np = np.ma.array(
+      np.zeros([bounds.raster_size_x, bounds.raster_size_y, 2], dtype=float),
+      mask=np.ones([bounds.raster_size_x, bounds.raster_size_y, 2], dtype=bool))
+
+  for x_idx, x in enumerate(tqdm(np.arange(bounds.minx, bounds.maxx, bounds.pixel_size_x, dtype=float))):
+    rows = []
+    row_indexes = []
+    for y_idx, y in enumerate(np.arange(bounds.miny, bounds.maxy, -bounds.pixel_size_y, dtype=float)):
+      # Row should contain all the features needed to predict, in the same
+      # column order the model was trained.
+      row = {}
+      row["lat"] = y
+      row["long"] = x
+
+      # Surround in try/except as we will be trying to fetch out of bounds data.
+      try:
+        if geometry_mask and pd.isnull(geometry_mask.value_at(x, y)):
+          continue
+        for geotiff_label, geotiff in geotiffs.items():
+          row[geotiff_label] = geotiff.value_at(x, y)
+          if pd.isnull(row[geotiff_label]):
+            raise ValueError
+      except (ValueError, IndexError):
+        continue # masked and out-of-bounds coordinates
+
+      rows.append(row)
+      row_indexes.append((y_idx,0,))
+
+    if (len(rows) > 0):
+      X = pd.DataFrame.from_dict(rows)
+      predictions = model.predict_on_batch(X)
+
+      means_np = predictions[:, 0]
+      vars_np = predictions[:, 1]
+      for mean, var, (y_idx, month_idx) in zip(means_np, vars_np, row_indexes):
+        predicted_np.mask[x_idx, y_idx, 0] = False # unmask since we have data
+        predicted_np.mask[x_idx, y_idx, 1] = False # unmask since we have data 
+        predicted_np.data[x_idx, y_idx, 0] = mean
+        predicted_np.data[x_idx, y_idx, 1] = var     
+
+  return predicted_np
+
+def dispatch_rasters(
+    required_rasters: typing.List[str],
+    use_earth_engine_assets : bool=False,
+    local_fallback: bool=False):
+  """
+  dispatch_rasters function
+  --------------------------------------------------
+  Loads rasters either from Google Earth Engine or from a local/gdrive
+  depending on parameters
+  --------------------------------------------------
+  Parameters:
+  required_rasters:
+    The list of rasters to load, e.g. ["DEM", "PA", "VPD"]
+  use_earth_engine_assets: 
+    Whether to load the rasters from earth engine. If False, loads from local/gdrive.
+  local_fallback:
+    If use_earth_engine_assets=True, whether to fallback to gdrive if rasters could not
+    be found in ee. 
+  """
+
+  # Print a warning if the raster's projection is now WGS 84.
+  def check_proj(geotiff: raster.AmazonGeoTiff):
+    projection = geotiff.gdal_dataset.GetProjection()
+    geogcs = osr.SpatialReference(wkt=projection).GetAttrValue('geogcs')       
+    if geogcs != 'WGS 84':
+      print(f"{_WARNING_COLOR}WARNING: {geogcs} projections will soon no longer be supported. "
+            f"Please reproject to WGS 84 instead{_ENDC}")
```

Comment:
Why is this a nested method as opposed to the alternative of leaving it at module-scope? Is there anywhere else that we handle a list of rasters? If so, that may also be an opportunity to use this functionality. Otherwise, this probably makes sense. Curious what you have in mind.

Reply:
I can't see it being used anywhere else, but i moved it to module-level scope to be consistent with the rest of the file.